ai-decision-services/ai-ad-dpp-engine/prepare-config-data-and-app/prepare-config-data-and-app.md

```diff
@@ -0,0 +1,217 @@
+*   For oracle, you will require:
+    *   **adbId** \- the OCID of your ADW application. 
+    *   **tableName** \- name of the table. 
+    *   **connectionId** \- the format is **<DBName>\_high**, which can be found under tnsnames.ora of [Wallet](https://docs.oracle.com/en/cloud/paas/autonomous-database/adbsa/connect-download-wallet.html#GUID-B06202D2-0597-41AA-9481-3B174F75D4B1). 
+    *   **user** \- the owner of the table. In most cases it will be **admin**. 
+    *   **password** \- the password you setup when you download the Wallet. 
+
+## phaseInfo
+
+The flag indicating whether it is a training or inferencing phase, plus other required input parameters. Here we will include an object storage connector, the workflow will use it to store or read metadata, which is useful for data dependent steps (for example, one-hot encoding).
+
+*   **connector** - the connector to object storage. It will be composed of
+    *   **namespace** - the namespace of the target object storage bucket. Required if it is for object storage.
+    *   **bucket** - the bucket name. Required if it is for object storage.
+    *   **objectName** \- the full path and name of the object. Required if it is for object storage.
+
+## processingSteps
+
+This will be a list of all the transformations you want to apply. This is the core of the data processing. Every item in the list is a step, and all the steps will be executed in sequence, one after another. Users should make sure the sequence makes sense, or exceptions will be thrown. DPP doesn't responsible for process validation before execution. For each step, you need to specify the following.
+
+*   **stepType** \- can be either **singleDataFrameProcessing** (for transforming on top of single dataframe) or **combineDataFrames** (for merging or joining two dataframes).
+*   **configurations** - the configuration related to the step you take, which will include:
+    *   **dataframeName** - a name you given to the output of the transformation.
+    *   **steps** - a list of all transformers you want to apply to your target dataframe at this time. For each step, there will be:
+        *   **stepName** - the name of the transformer you want to apply. It must exactly matches the one you defined in archive. For the full list of offered transformers, please check **Appendix II: List of Transformers**.
+        *   **args** \- all the arguments required for the transformer you choose to use. For details about it, check **Appendix II: List of Transformers**.
+
+## stagingDestination
+
+This is where we store intermediate processed data for post processing. Basically these will be dataframes stored in csv format at an assigned object storage bucket. 
```

Comment:
Passive tone again - This is the location where intermediate results will be stored before post processing steps begin ..... (Reword as appropriate)